primitives/trie/src/storage_proof.rs

```diff
@@ -54,6 +54,11 @@ impl StorageProof {
 		self.trie_nodes.is_empty()
 	}
 
+	/// Returns `true` if the proof is equivalent to the other one.
+	pub fn compare(&self, other: &Self) -> bool {
+		self.trie_nodes == other.trie_nodes
+	}
```

Comment:
This makes no sense. Now you are ordering them again before adding them here. I mean that is fine, but then we should do that in the constructor and can just bring back `PartialEq` for everything. You `compare` method here is essentially the exact same as the `PartialEq` implementation.

Reply:
Hmm, this was intended to serve as a convenient method for the cumulus companion where `PartialEq` can not be avoided as `ParachainInherentData` is used in one pallet Call. I'm kind of confused, what's your suggestion here then?

Reply:
I mean, I was thinking that `compare` would sort the values on demand. What you have here now is actually the same as using `BTreeSet`. 

Sorry! Maybe just revert the last commits, we take the `BTreeSet` in the storage proof and call it a day.

Reply:
But using `BTreeSet` will break the existing code as mentioned by @cheme above -_- Let's just sort the trie nodes before converting to the storage proof and keep `PartialEq`?

Reply:
The way scale-codec work BTreeSet do not break it, my comment was a bit unclear (it does Not check order when decoding so it is fine).
(of course I am talking of the StorageProof not the Compact one)

Reply:
Sorry, I'm lost. How to deal with the failed test `warp_sync` if using `BTreeSet` https://github.com/paritytech/substrate/pull/10915#issuecomment-1050734425? Or do you mean we can use `BTreeSet` for `StorageProof` and still use `Vec` for `CompactProof`?

Reply:
@bkchr PR updated, now it's time to take care of the cumulus companion :D How would you like the `PartialEq` implemented in https://github.com/paritytech/cumulus/pull/1051/files?